3rdParty/s2geometry/0e7b146/src/s2/util/gtl/compact_array.h

```diff
@@ -413,8 +413,10 @@ class compact_array_base {
     value_allocator_type allocator;
 
     T* new_ptr = allocator.allocate(capacity());
-    memcpy(new_ptr, Array(), old_capacity * sizeof(T));
-    allocator.deallocate(Array(), old_capacity);
+    if (old_capacity == 0) {
```

Comment:
I don't understand this change. Shouldn't it be `if (old_capacity != 0)` here?

Reply:
Thanks!